src/components/questionnaire.js

```diff
@@ -0,0 +1,224 @@
+import React, { Component } from 'react';
+import '../styles/conversation.css';
+import Question from '../components/question';
+import Validator from '../core/ci-validation';
+
+class QuestionnaireComponent extends Component {
+    constructor(props) {
+        super(props);
+        
+        this.state = this.getInitialState();
+        this.state.messages = this.props.messages;
+	};
+
+	componentWillMount() {
+        this.startQuestionnaire();
+    };
+
+	getInitialState = () => {
+        return {
+            activeMessage: {},
+			answers: {},
+			validationErrors: [],
+            log: [],
+			isLoading: false,
+			messages: this.props.messages
+        };
+	};
+	
+	addToLog = (item) => {
+		let log = this.state.log;
+		log.push(item);
+		this.setState({
+			log: log
+		});
+	};
+
+	getLogById = (id) => {
+		return this.state.log.filter(item => Object.hasOwnProperty.call(item, 'id') ? item.id === id : false)[0];
+	};
+
+	getMessageById = (id) => {
+		if (!id) {
+			return this.state.messages[0];
+		}
+		return this.state.messages.filter(message => message.id === id)[0];
```

Comment:
Same comment here -- index 0 might not exist.

Reply:
addressed